pkg/object/object.go

```diff
@@ -70,10 +75,36 @@ type Volume struct {
 	StorageClass *string
 }
 
+type Secret struct {
+	Name string
+	Data []SecretData
+}
+
+type SecretData struct {
+	Key       string
+	Plaintext *string
+	Base64    *string
+	File      *string
+}
+
 type OpenCompose struct {
 	Version  string
 	Services []Service
 	Volumes  []Volume
+	Secrets  []Secret
+}
+
+// This struct is not a part of the OpenCompose spec, but this is used to
+// store the data input metadata which is used in other parts of the code
+// to make decisions.
+// For instance, this is being set in cmd.GetValidatedObject(), and getting
+// in v1.Decode() to convert the relative path of the secret file from the
+// OpenCompose file, and convert it to an absolute path
+type Input struct {
+	Data     []byte
+	STDIN    bool
+	URL      *string
+	FilePath *string
```

Comment:
We have done for others specifically in `encoding.go` so that we can find what is optional and what is not, is that a case with this particular struct? Since this is populated by us and not the user input.